src/imap.rs

```diff
@@ -2103,6 +2102,18 @@ async fn get_modseq(context: &Context, folder: &str) -> Result<u64> {
         .unwrap_or(0))
 }
 
+/// Compute the imap search expression for all self-sent mails (for all self addresses)
+pub(crate) async fn get_imap_self_sent_search_command(context: &Context) -> Result<String> {
+    // See https://www.rfc-editor.org/rfc/rfc3501#section-6.4.4 for syntax of SEARCH and OR
+    let mut search_command = format!("FROM {}", context.get_primary_self_addr().await?);
+
+    for item in context.get_secondary_self_addrs().await? {
+        search_command = format!("OR ({}) (FROM {})", search_command, item);
+    }
+
+    Ok(search_command)
+}
```

Comment:
was this ever tested or do we have a python test for this?

Reply:
I tested it manually, and it works even for secondary addrs now. We also have a Python test that tests that `add_all_recipients_as_contacts()` works, `test_fetch_existing()` (I just double-checked that it fails when I comment out `add_all_recipients_as_contacts()`